packages/flutter/lib/src/material/ink_decoration.dart

```diff
@@ -0,0 +1,356 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material.dart';
+
+/// A convenience widget for drawing images and other decorations on [Material]
+/// widgets, so that [InkWell] and [InkResponse] splashes will render over them.
+///
+/// Ink splashes and highlights, as rendered by [InkWell] and [InkResponse],
+/// draw on the actual underlying [Material], under whatever widgets are drawn
+/// over the material (such as [Text] and [Icon]s). If an opaque image is drawn
+/// over the [Material] (as with a [Container] or [DecoratedBox]), these ink
+/// effects will not be visible, as they will be entirely obscured by the opaque
+/// graphics drawn above the [Material].
+///
+/// This widget draws the given [Decoration] directly on the [Material], in the
+/// same way that [InkWell] and [InkResponse] draw there. This allows the
+/// splashes to be drawn above the otherwise opaque graphics.
+///
+/// An alternative solution is to use a [MaterialType.transparency] material
+/// above the opaque graphics.
```

Comment:
... because in that case ink well and ink response descendants will draw their splashes on the otherwise transparent material

Reply:
done